src/db/delete.rs

```diff
@@ -16,20 +16,22 @@ enum CrateDeletionError {
 pub fn delete_crate(conn: &Connection, name: &str) -> Result<(), Error> {
     let crate_id = get_id(conn, name)?;
     delete_crate_from_database(conn, name, crate_id)?;
-    if let Some(s3) = s3_client() {
+
+    if let Some(backend) = s3_client().map(|client| S3Backend::new(client, S3_BUCKET_NAME)) {
```

Comment:
That'd also require taking in a bucket name, so it'd be changing calls in a slightly different way

Reply:
Do we need to take in the bucket name? Can we pass `S3_BUCKET_NAME` from `s3_client`? It seems silly to pass a constant that's always the same.

Reply:
We need to take the bucket name at some point [due to tests](https://github.com/rust-lang/docs.rs/blob/master/src/storage/s3/test.rs#L14-L24) (Note that I've removed the `Box::leak` in my PR since that's not good)